### ISSUE TITLE
New languages?

### ISSUE THREAD

**Issue by HoratiuMl** [zaps166/NFSIISE#25]
Hello.

Can new languages be implemented into the game?

If yes, then can you please offer some info on how is it done? I'd like to work on a romanian translation.

**Comment by zaps166** [comment]
> Can new languages be implemented into the game?

Yes, but it won't be easy. Translations can be found in `Need For Speed II SE` directory as `text.*`, but it is binary file, you must guess what is inside and modify in hex-editor (source is in assembly, so it is difficult to understand it) :)

**Comment by HoratiuMl** [comment]
After searching the internet for hours upon hours, trying to find a tool called "NFSStrings" but only finding broken links from the early 2000s... I come back here victorious!

I have found it, finally: http://web.archive.org/web/20080623221127/http://nfsgb.nd4spdworld.com/zip/nfs3/files/nfsstring.zip

And it is working!

Currently I tried replacing the text.eng file and it seems to be working fine. However, I'm not yet sure how to add it as a separate, new language. I will try more when I get home and have access to my desktop.

What worries me is that some of the text is stored as image files, but this is rare. While creating new images wouldn't be hard, especially since I already plan to recreate most assets of the game (at least those that are within my reach), I don't know how would I make the game actually use them. Though... I would be happy even with just the language file, because as I said, almost everything is in there.

Also the voice acting is going to be a problem - that one's gonna stay english until someone decides to record new files - but that's incredibly unlikely.
